Source/DataSources/GeometryHeightProperty.js

```diff
@@ -0,0 +1,201 @@
+     *     ellipse : {
+     *         semiMinorAxis : 300.0,
+     *         semiMajorAxis : 200.0,
+     *         material : Cesium.Color.BLUE,
+     *         height : new Cesium.GeometryHeightProperty(20.0, Cesium.HeightReference.RELATIVE_TO_GROUND)
+     *     }
+     * });
+     */
+    function GeometryHeightProperty(height, heightReference) {
+        this._height = undefined;
+        this._heightSubscription = undefined;
+        this._heightReference = undefined;
+        this._heightReferenceSubscription = undefined;
+        this._definitionChanged = new Event();
+
+        this._resultObject = {
+            height: 0,
+            heightReference: HeightReference.NONE
+        };
+
+        this.height = height;
+        this.heightReference = heightReference;
+    }
+
+    defineProperties(GeometryHeightProperty.prototype, {
+        /**
+         * Gets a value indicating if this property is constant.
+         * @memberof GeometryHeightProperty.prototype
+         *
+         * @type {Boolean}
+         * @readonly
+         */
+        isConstant : {
+            get : function() {
+                return Property.isConstant(this._height) && Property.isConstant(this._heightReference);
+            }
+        },
+        /**
+         * Gets the event that is raised whenever the definition of this property changes.
+         * @memberof GeometryHeightProperty.prototype
+         *
+         * @type {Event}
+         * @readonly
+         */
+        definitionChanged : {
+            get : function() {
+                return this._definitionChanged;
+            }
+        },
+        /**
+         * Gets or sets the property used to compute the height.
+         * @memberof GeometryHeightProperty.prototype
+         *
+         * @type {Property}
+         */
+        height : createPropertyDescriptor('height'),
+        /**
+         * Gets or sets the property used to define the height reference.
+         * @memberof GeometryHeightProperty.prototype
+         *
+         * @type {Property}
+         */
+        heightReference : createPropertyDescriptor('heightReference')
+    });
+
+    /**
+     * Gets the minimum terrain height based on the positions at the provided time.
+     *
+     * @param {JulianDate} time The time for which to retrieve the value.
+     * @returns {Number} The minimum terrain height
+     */
+    GeometryHeightProperty.prototype.getValue = function(time) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('time', time);
+        //>>includeEnd('debug');
+
+        var heightReference = Property.getValueOrDefault(this._heightReference, time, HeightReference.NONE);
+        var height;
+        if (heightReference !== HeightReference.CLAMP_TO_GROUND) {
+            height = Property.getValueOrDefault(this._height, time, 0);
+        } else {
+            height = 0;
+        }
+
+        var result = this._resultObject;
```

Comment:
As discussed offline, to not break the interface we can do `height = new Number(height)` here and glob `heightReference` onto that.  I'm not crazy about it, but it will still be cleaner than what we're doing now.